Miscellaneous fixes in version numbers.

sjrd opened the pull request.

sjrd commented:
Review by @gzm0

scala-jenkins commented:
Refer to this link for build results (access rights to CI server needed): 

<a href='https://scala-webapps.epfl.ch/jenkins/job/scalajs-pr/1809/'>https://scala-webapps.epfl.ch/jenkins/job/scalajs-pr/1809/</a>
<a href='https://scala-webapps.epfl.ch/jenkins/job/scalajs-matrix-build/2348/'>https://scala-webapps.epfl.ch/jenkins/job/scalajs-matrix-build/2348/</a>
Test PASSed.

gzm0 commented on ir/src/main/scala/org/scalajs/core/ir/ScalaJSVersions.scala:
```diff
@@ -2,19 +2,26 @@ package org.scalajs.core.ir
+   */
+
+  /** Scala.js version. */
+  val current = "0.6.4-SNAPSHOT"
```
Missing return type.

gzm0 commented on ir/src/main/scala/org/scalajs/core/ir/ScalaJSVersions.scala:
```diff
@@ -2,19 +2,26 @@ package org.scalajs.core.ir
-  /** true iff the Scala.js version of this build is a snapshot version. */
-  final val currentIsSnapshot = current endsWith "-SNAPSHOT"
+  /** true iff the Scala.js version is a snapshot version. */
+  val currentIsSnapshot = current endsWith "-SNAPSHOT"
```
Same

gzm0 commented on ir/src/main/scala/org/scalajs/core/ir/ScalaJSVersions.scala:
```diff
@@ -2,19 +2,26 @@ package org.scalajs.core.ir
    *  - `current`
    */
-  final val binaryEmitted = current
+  val binaryEmitted = current
```
Same

gzm0 commented:
LGTM otherwise.

gzm0 commented:
(It feels like we had the `MinorSnapshotVersion` thingy before... strange we removed it)

gzm0 commented:
I assume you tested the regex?

sjrd commented:
> I assume you tested the regex?

Yes I did ^^

> (It feels like we had the MinorSnapshotVersion thingy before... strange we removed it)

We removed in commit 216f2faabd7a1eba42aa7e400ed5b6096ed43ece, probably because we thought that due to using `binaryEmitted`, we would not have this problem anymore. But that was without taking into account that, at times, even `binaryEmitted` is a minor snapshot.

sjrd commented:
Updated.

scala-jenkins commented:
Refer to this link for build results (access rights to CI server needed): 

<a href='https://scala-webapps.epfl.ch/jenkins/job/scalajs-pr/1820/'>https://scala-webapps.epfl.ch/jenkins/job/scalajs-pr/1820/</a>
<a href='https://scala-webapps.epfl.ch/jenkins/job/scalajs-matrix-build/2360/'>https://scala-webapps.epfl.ch/jenkins/job/scalajs-matrix-build/2360/</a>
Test PASSed.

gzm0 commented:
LGTM
